enzyme/Enzyme/Utils.cpp

```diff
@@ -158,6 +158,98 @@ Function *getOrInsertDifferentialFloatMemcpy(Module &M, PointerType *T,
   return F;
 }
 
+Function *getOrInsertDifferentialMemcpyStrided(Module &M, PointerType *T,
+                                               unsigned dstalign,
+                                               unsigned srcalign,
+                                               Value *stride) {
+  Type *elementType = T->getElementType();
+  assert(elementType->isFloatingPointTy());
+  std::string name = "__enzyme_memcpy_" + tofltstr(elementType) + "da" +
+                     std::to_string(dstalign) + "sa" +
+                     std::to_string(srcalign) + "stride" +
+                     std::to_string(cast<ConstantInt>(stride)->getSExtValue());
+  FunctionType *FT =
+      FunctionType::get(Type::getVoidTy(M.getContext()),
+                        {T, T, Type::getInt32Ty(M.getContext())}, false);
+
+#if LLVM_VERSION_MAJOR >= 9
+  Function *F = cast<Function>(M.getOrInsertFunction(name, FT).getCallee());
+#else
+  Function *F = cast<Function>(M.getOrInsertFunction(name, FT));
+#endif
+
+  if (!F->empty())
+    return F;
+
+  F->setLinkage(Function::LinkageTypes::InternalLinkage);
+  F->addFnAttr(Attribute::ArgMemOnly);
+  F->addFnAttr(Attribute::NoUnwind);
```

Comment:
Probably also worth adding readonly to src and writeonly (if sufficient llvm version) to dst.